src/shared_utilities/utils.jl

```diff
@@ -20,21 +20,72 @@ Helper function used in computing tendencies of vertical diffusion terms.
 """
 to_scalar_coefs(vector_coefs) = map(vector_coef -> vector_coef.u₃, vector_coefs)
 
+
+"""
+    add_dss_buffer_to_aux(p, domain::Domains.AbstractDomain)
+
+Fallback method for `add_dss_buffer_to_aux` which does not add a dss buffer.
+"""
+add_dss_buffer_to_aux(p, domain::Domains.AbstractDomain) = p
+
+"""
+    add_dss_buffer_to_aux(
+        p,
+        domain::Union{Domains.Plane, Domains.SphericalSurface},
+    )
+
+Adds a dss buffer corresponding to `domain.space` to `p` with the name `dss_buffer_2d`, 
+appropriate for a 2D domain.
```

Comment:
could you add a comment explaining why we store `dss_buffer_2d` and `dss_buffer_3d` separately and with different names?

Reply:
"We use a name which specifically denotes that the buffer is on a 3(2)d space
because for some models, we require both a buffer on the 3d space as well as on the surface 
2d space, because some models have prognostic variables that are only defined on the surface 
space."